### PR TITLE
Add option to `Transforms.ThreeQubitSquash` to target TK2 gates

### PR DESCRIPTION
**Pull request by cqc-alec**
The existing `three_qubit_synthesis()` method decomposes a unitary into at most 20 CX gates. I have added an analogous `three_qubit_tk_synthesis()` method that decomposes a unitary into at most 15 TK2 gates. (It may be possible to do better than this, but I don't know how. I simply took the existing steps based on the cosine--sine decomposition and modified them. I had hoped that it might be possible to use fewer gates in the `two_qubit_diag_adjoint_plex_tk()` and `two_qubit_modified_cossin_circ_tk()` subroutines, but numerical experiments suggest not.)

Decompose `Unitary3qBox` using this method.

Add an option to the `three_qubit_squash` transform (and its Python wrapper) to target TK2 instead of CX.

### CODE REVIEW

File: tket/src/Transformations/ThreeQubitSquash.cpp [hunk 1]
```diff
@@ -108,22 +108,28 @@ class QInteraction {
...
-    // when necessary.
-    normalise_TK2().apply(repl);
-    decompose_TK2().apply(repl);
-    clifford_simp(false).apply(repl);
-    return repl;
+    if (target_2q_gate == OpType::CX) {
+      Circuit repl = three_qubit_synthesis(get_3q_unitary(circ));
+      normalise_TK2().apply(repl);
+      decompose_TK2().apply(repl);
+      clifford_simp(false).apply(repl);
```
Comment (lmondada) [inline tket/src/Transformations/ThreeQubitSquash.cpp:125]: If we `squash_1qb_to_tk1` in the `TK2` case, would it not make sense to do it here too? I would also move out `three_qubit_synthesis(get_3q_unitary(circ))` from the if-else clause to highlight only the differences between the two OpTypes, but that's subjective taste.

**Review by lmondada — CHANGES_REQUESTED** [review]
Looking good, only very minor things. I've added some comments about conjugating with reversible circuits to get fewer CX counts in some cases, but that was more a note to self. I will add an issue to be investigated later.

File: pytket/binders/transform.cpp [hunk 2]
```diff
@@ -327,8 +327,9 @@ PYBIND11_MODULE(transform, m) {
           py::arg("cx_fidelity") = 1.)
       .def_static(
           "ThreeQubitSquash", &Transforms::three_qubit_squash,
-          "Squash three-qubit subcircuits into subcircuits having fewer CX "
-          "gates, when possible.")
+          "Squash three-qubit subcircuits into subcircuits having fewer "
+          "2-qubit gates of the target type (default CX), when possible.",
+          py::arg("target_2q_gate") = OpType::CX)
```
[the remark this answers is not in the document]
Comment (cqc-alec) [reply pytket/binders/transform.cpp:343]: I'll change to `target_2qb_gate` throughout.

**Review by lmondada — APPROVED** [review]
Thank you!

**Pull request merged**